octoprint_matrix_notifier/plugin.py

```diff
@@ -286,20 +271,67 @@ def send_snapshot(self):
         """
         Capture and then send a snapshot from the camera.
         """
-        file_path = self.capture_snapshot()
-
-        with open(file_path, "rb") as fobj:
-            data = fobj.read()
+        data = self.take_image()
 
         mxc_url = self.client.upload_media(data, "image/jpg")["content_uri"]
 
         img_w, img_h = get_image_size_from_bytesio(io.BytesIO(data), len(data))
 
         content = {
             "msgtype": "m.image",
-            "body": file_path.name,
+            "body": time.strftime("%Y_%m_%d-%H_%M_%S") + ".jpg",
             "info": {"mimetype": "image/jpg", "w": img_w, "h": img_h},
             "url": mxc_url,
         }
 
         return self.client.room_send(self.room_id, "m.room.message", content)
+
+    def getProxies(self):
+        http_proxy = self._settings.get(["http_proxy"])
+        https_proxy = self._settings.get(["https_proxy"])
+        return {"http": http_proxy, "https": https_proxy}
+
+    def take_image(self, snapshot_url=""):
+        if snapshot_url == "":
+            snapshot_url = self._settings.global_get(["webcam", "snapshot"])
+        if snapshot_url == "":
```

Comment:
I think this is a little cleaner:

```suggestion
    def take_image(self, snapshot_url=None:
        if not snapshot_url:
            snapshot_url = self._settings.global_get(["webcam", "snapshot"])
        if not snapshot_url:
```

Reply:
update the branch with the multicam support changes which refactor this place ;)